django/contrib/postgres/fields/ranges.py

```diff
@@ -132,6 +133,44 @@ def db_type(self, connection):
 RangeField.register_lookup(lookups.Overlap)
 
 
+class DateTimeRangeContains(models.Lookup):
+    lookup_name = 'contains'
+
+    def as_sql(self, compiler, connection):
+        # Transform rhs value for db lookup
+        value = self.rhs
+        if isinstance(value, datetime.datetime):
+            output_field = models.DateTimeField()
+        elif isinstance(value, datetime.date):
+            output_field = models.DateField()
+        else:
+            output_field = None
+
+        if output_field:
+            value = models.Value(value, output_field=output_field)
+            self.rhs = value.resolve_expression(compiler.query)
```

Comment:
@charettes thanks for review. I think that the most appropriate place will be `process_rhs` method. I reorganized PR.